src/components/Capture/withCrossDeviceWhenNoCamera.js

```diff
@@ -34,7 +35,7 @@ export default WrappedComponent =>
         }
         const step = 0
         const excludeStepFromHistory = true
-        changeFlowTo('crossDeviceSteps', step, excludeStepFromHistory)
+        isDesktop && changeFlowTo('crossDeviceSteps', step, excludeStepFromHistory)
```

Comment:
The logic here was pretty tangled even before your change. I would follow this method approach to end the execution instead of introducing another pattern. `if(isDesktop) { return; }`